fluence-healthcheck/src/index.js

```diff
@@ -0,0 +1,72 @@
+import "@fluencelabs/js-client.node";
+import * as core from "@actions/core";
+import { Fluence } from "@fluencelabs/js-client.api";
+import * as network from "@fluencelabs/fluence-network-environment";
+import { checkPeer } from "./_aqua/main.js";
+
+async function main() {
+  const env = core.getInput("env");
+  const timeout = parseInt(core.getInput("timeout"));
+  let peers, relay;
+  console.log(process.env.GITHUB_CONTEXT);
+  console.log(env);
+
+  const errorMessages = [];
+
+  switch (env) {
+    case "stage":
+      peers = network.stage;
+      relay = network.randomStage();
+      break;
+    case "testnet":
+      peers = network.testNet;
+      relay = network.randomTestNet();
+      break;
+    case "kras":
+      peers = network.kras;
+      relay = network.randomKras();
+      break;
+    default:
+      console.error('Invalid "env" input. Use "stage", "testnet", or "kras".');
+      process.exit(1);
+  }
+
+  try {
+    await Fluence.connect(relay.multiaddr);
+
+    const data = peers.map(({ peerId, multiaddr }, index) => ({
+      peer: peerId,
+      multiaddr: multiaddr,
+      validators: peers.filter((_, idx) => idx !== index).map((n) => n.peerId),
+    }));
+
+    for (const setup of data) {
+      try {
+        console.log(`Checking peer: ${setup.multiaddr}`);
+        const result = await checkPeer(setup.peer, setup.validators, timeout, {
```

Comment:
I think it it important to check `result` to contain only `TARGET REACHABLE` and fail on anything else

Reply:
i don't see how `try catch` will detect problems given that Aqua code has built-in timeout and should always return result